### PR TITLE
Bugfix - Missing Password

### PR DESCRIPTION
**Pull request by sakshibobade21**
## Proposed changes

This PR addresses Issue:
Adds a dialog for entering a password when the resume button is clicked to fixe the bug where the password was missing from the PASS command.

This PR depends upon the following PRs:

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Change in a documentation
- [ ] Refactor the code 
- [ ] Chore, repository cleanup, updates the dependencies.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## PR Checklist
Please delete options that are not relevant.
- [ ] If the changes in this PR are meant for the next release / mainline, this PR targets a "staging" branch.
- [ ] My code follows the style guidelines of this project (see: [Contributing guideline](https://github.com/zowe/zen/blob/v2.x/staging/CONTRIBUTING.md))
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] video or image is included if visual changes are made

- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works, or describe a test method below

### CODE REVIEW

File: src/renderer/components/common/passwordDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
+/*
```
Comment (timgerstel) [inline src/renderer/components/common/passwordDialog.tsx:1]: Should probably reuse the connection component instead of adding a new one

**Comment by timgerstel** [comment]
Im confused on what this PR does. When you click the resume feature, we already show the connection screen if there is no valid session. So this PR now adds...a second login screen?

**Comment by sakshibobade21** [comment]
@timgerstel 

https://github.com/zowe/zen/issues/184
https://jira.rocketsoftware.com/browse/MVD-6437

This dialog is specifically for resuming within the same session. The details you mentioned above pertain to resuming in a different session.

As we discussed before, we aim to avoid saving the password, even within the redux state, correct @1000TurquoisePogs  ?

Therefore, when the user clicks 'Save and Close,' we will remove the password. For resuming within the same session, only the password is required, not the other fields from the connection tab. Hence, the purpose of this dialog

File: src/renderer/components/common/passwordDialog.tsx (hunk 1 above)
Comment (DivergentEuropeans) [reply src/renderer/components/common/passwordDialog.tsx:1]: @timgerstel I think you are fundamentally misunderstanding what this PR is about

Comment (DivergentEuropeans) [reply src/renderer/components/common/passwordDialog.tsx:1]: Ah, I see

>  When you click the resume feature, we already show the connection screen if there is no valid session. 

This part was not working for same session. Sakshi fixed this + added more minimal UI

**Pull request merged**